### PR TITLE
Implement alarm apis

### PR DESCRIPTION
**Pull request by chris-sun-star**
## Summary

Implement alarm apis
Integrate alertmanager into oceanbase-dashboard chart
change make command of dashboard starts with dashboard-

### CODE REVIEW

File: internal/dashboard/handler/alarm_handler.go [hunk 1]
```diff
@@ -257,7 +292,7 @@ func DeleteReceiver(_ *gin.Context) (any, error) {
 // @Router /api/v1/alarm/receiver/templates [POST]
 // @Security ApiKeyAuth
 func ListReceiverTemplates(_ *gin.Context) ([]receiver.Template, error) {
-	return nil, httpErr.NewNotImplemented("not implemented")
+	return alarm.ListReceiverTemplates()
```
Comment (powerfooI) [inline internal/dashboard/handler/alarm_handler.go:295]: The `context.Context` parameter should be passed in.

File: internal/dashboard/model/alarm/alert/alert.go [hunk 2]
```diff
@@ -13,27 +13,96 @@ See the Mulan PSL v2 for more details.
...
+	Labels      []common.KVPair       `json:"labels,omitempty"`
+	Summary     string                `json:"summary,omitempty"`
+	Description string                `json:"description,omitempty"`
+}
+
+func NewAlert(alert *ammodels.GettableAlert) (*Alert, error) {
+	rule, ok := alert.Labels[alarmconstant.LabelRuleName]
+	if !ok {
+		// TODO: return error
+		// return nil, errors.New("Convert alert failed, no rule")
```
Comment (powerfooI) [inline internal/dashboard/model/alarm/alert/alert.go:52]: This line can be uncommented. This error handling is required I thought.

File: internal/dashboard/model/alarm/external/types.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+You may obtain a copy of Mulan PSL v2 at:
+
+	http://license.coscl.org.cn/MulanPSL2
+
+THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
+EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
+MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
+See the Mulan PSL v2 for more details.
+*/
+package external
```
Comment (powerfooI) [inline internal/dashboard/model/alarm/external/types.go:14]: This file seems to be unused. Can we delete it?

File: internal/dashboard/model/alarm/receiver/enums.go [hunk 4]
```diff
@@ -17,6 +17,17 @@ type ReceiverType string
 
 // TODO add all types
```
Comment (powerfooI) [inline internal/dashboard/model/alarm/receiver/enums.go:18]: This TODO seems to have been done

File: internal/dashboard/model/alarm/silence/enums.go [hunk 5]
```diff
@@ -16,5 +16,7 @@ type State string
 
 // TODO add all states
```
Comment (powerfooI) [inline internal/dashboard/model/alarm/silence/enums.go:17]: This TODO could be deleted.

File: internal/dashboard/router/router.go [hunk 6]
```diff
@@ -63,13 +63,17 @@ func InitRoutes(router *gin.Engine) {
 		router.Use(static.Serve("/api-gen", static.LocalFile("internal/dashboard/generated/swagger", false)))
 	}
 
+	v1Group := router.Group("/api/v1")
 	// login api does not require login
-	v1Group := router.Group("/api/v1",
-		middleware.LoginRequired(),
-		middleware.RefreshExpiration(),
-	)
+	if os.Getenv("ENABLE_DEV_MODE") != "true" {
```
Comment (powerfooI) [inline internal/dashboard/router/router.go:68]: There is an old environment named `DEBUG_DASHBOARD` may be reused here.

File: internal/assets/receiver_templates/email_config.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+to: "your email address"
+smarthost: 
+  host: "address of your email provider"
+  port: "port of your email provider"
+auth_username: "username"
+auth_password: "password"
+# auth_secret: "secret"
```
[the remark this answers is not in the document]
Comment (chris-sun-star) [reply internal/assets/receiver_templates/email_config.yaml:7]: No, I write the comments in purpose, the user can config the receiver by removing the '#' and config value

File: internal/assets/receiver_templates/victorops_config.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+api_url: 'your api url'
+routing_key: 'YOUR_ROUTING_KEY'
+message_type: '{{ .CommonLabels.severity }}'
```
[the remark this answers is not in the document]
Comment (chris-sun-star) [reply internal/assets/receiver_templates/victorops_config.yaml:3]: go template is the general real configuration for victorops

File: internal/dashboard/business/alarm/silencer.go [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+	postableSilence := &ammodels.PostableSilence{
+		ID:      param.Id,
+		Silence: silencer,
+	}
+	okBody := amsilence.PostSilencesOKBody{}
+	resp, err := getClient().R().SetContext(ctx).SetHeader("content-type", "application/json").SetBody(postableSilence).SetResult(&okBody).Post(fmt.Sprintf("%s%s", alarmconstant.AlertManagerAddress, alarmconstant.MultiSilencerUrl))
+	if err != nil || resp.StatusCode() != http.StatusOK {
+		return nil, errors.Wrap(err, errors.ErrExternal, "Query silencers from alertmanager")
+	}
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (chris-sun-star) [reply internal/dashboard/business/alarm/silencer.go:141]: should delete line 138 to 140

File: internal/dashboard/business/alarm/receiver.go [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+		configReceivers = append(configReceivers, amreceiver)
+	}
+	if !foundReceiver {
+		return errors.NewBadRequest(fmt.Sprintf("Receiver %s not exists", name))
+	}
+	config.Receivers = configReceivers
+	return updateAlertManagerConfig(ctx, config)
+}
+
+func ListReceiverTemplates() ([]receiver.Template, error) {
```
[the remark this answers is not in the document]
Comment (chris-sun-star) [reply internal/dashboard/business/alarm/receiver.go:117]: these two function returns templates directly from bindata, it's not necessary to pass a context

File: internal/dashboard/handler/alarm_handler.go (hunk 1 above)
Comment (chris-sun-star) [reply internal/dashboard/handler/alarm_handler.go:295]: there's no need to pass the context

File: internal/dashboard/business/alarm/receiver.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (chris-sun-star) [reply internal/dashboard/business/alarm/receiver.go:117]: these two function returns templates directly from bindata, it's not necessary to pass a context

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**